tutorials/Inference-Bert-Model-for-High-Performance-with-ONNX-Runtime-on-AzureML.ipynb

```diff
@@ -0,0 +1,690 @@
+    "    # Retrieve best results from file.\n",
+    "    result = {}\n",
+    "    with open(output_prediction_file, \"r\") as f:\n",
+    "        result = json.load(f)\n",
+    "    return result\n",
+    "\n",
+    "def run(input_data_json):\n",
+    "    try:\n",
+    "        # load in our data\n",
+    "        input_ids, input_mask, segment_ids = preprocess(input_data_json)\n",
+    "        RawResult = collections.namedtuple(\"RawResult\", [\"unique_id\", \"start_logits\", \"end_logits\"])\n",
+    "        \n",
+    "        n = len(input_ids)\n",
+    "        bs = 1\n",
+    "        all_results = []\n",
+    "        start = time.time()\n",
+    "        for idx in range(0, n):\n",
+    "            item = all_examples[idx]\n",
+    "            # this is using batch_size=1\n",
+    "            # feed the input data as int64\n",
+    "            data = {\n",
+    "                    \"segment_ids\": segment_ids[idx:idx+bs],\n",
+    "                    \"input_ids\": input_ids[idx:idx+bs],\n",
+    "                    \"input_mask\": input_mask[idx:idx+bs]\n",
+    "                    }\n",
+    "            result = session.run([\"start\", \"end\"], data)\n",
+    "            in_batch = result[0].shape[0]\n",
+    "            start_logits = [float(x) for x in result[1][0].flat]\n",
+    "            end_logits = [float(x) for x in result[0][0].flat]\n",
+    "            for i in range(0, in_batch):\n",
+    "                unique_id = len(all_results)\n",
+    "                all_results.append(RawResult(unique_id=unique_id, start_logits=start_logits, end_logits=end_logits))\n",
+    "                \n",
+    "        end = time.time()\n",
+    "        print(\"total time: {}sec, {}sec per item\".format(end - start, (end - start) / len(all_results)))\n",
+    "        return {\"result\": postprocess(all_results),\n",
+    "                \"total_time\": end - start, \n",
+    "               \"time_per_item\": (end - start) / len(all_results)}\n",
+    "    except Exception as e:\n",
+    "        result = str(e)\n",
+    "        return {\"error\": result}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3.3 - Write Environment File\n",
+    "\n",
+    "We create a YAML file that specifies which dependencies we would like to see in our container."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.conda_dependencies import CondaDependencies \n",
+    "\n",
+    "myenv = CondaDependencies.create(pip_packages=[\"numpy\",\"onnxruntime\",\"azureml-core\", \"tensorflow\", \"wget\", \"pytorch_pretrained_bert\"])\n",
+    "\n",
+    "with open(\"myenv.yml\",\"w\") as f:\n",
+    "    f.write(myenv.serialize_to_string())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3.4 - Create Container Image\n",
```

Comment:
this does not look right. User shouldn't be creating containers. Also I can't find where Model.deploy is used. Please take a look at the AML examples.